dotcom-rendering/src/components/Card/Card.tsx

```diff
@@ -218,28 +210,65 @@ const CommentFooter = ({
 	);
 };
 
+type Media = {
+	type: CardImageType;
+};
+
+type MediaSlideshow = Media & {
+	type: 'slideshow';
+	slideshowImages: DCRSlideshowImage[];
+};
+
+type MediaAvatar = Media & {
+	type: 'avatar';
+	avatarUrl: string;
+};
+
+type MediaCrossword = Media & {
+	type: 'crossword';
+	imageUrl: string;
+};
+
+type MediaMainMedia = Media & {
+	type: 'mainMedia';
+	imageUrl: string;
+};
+
+type MediaVideo = Media & {
+	type: 'video';
+	mainVideo: CardYoutubeVideo;
+};
+
+type MediaUnion =
+	| MediaSlideshow
+	| MediaAvatar
+	| MediaCrossword
+	| MediaMainMedia
+	| MediaVideo;
```

Comment:
Is this helping or should we simply rely on inferred return types? I might refactor and opt for conciseness.

Reply:
Yep, all gone!